bc_obps/registration/api/operations.py

```diff
@@ -122,6 +134,16 @@ def list_operations(request, page: int = 1, sort_field: str = "created_at", sort
             Operation.objects.select_related("operator", "bc_obps_regulated_operation")
             .exclude(status=Operation.Statuses.NOT_STARTED)
             .exclude(status=Operation.Statuses.DRAFT)
+            .filter(
+                Q(bcghg_id__icontains=bcghg_id) if bcghg_id else Q(),
+                Q(bc_obps_regulated_operation__id__icontains=bc_obps_regulated_operation)
+                if bc_obps_regulated_operation
+                else Q(),
+                Q(name__icontains=name) if name else Q(),
+                Q(operator__legal_name__icontains=operator) if operator else Q(),
+                Q(status__icontains=status) if status else Q(),
+                Q(submission_date__icontains=submission_date) if submission_date else Q(),
+            )
```

Comment:
@Sepehr-Sobhani 

This is working for the multi-field search. What do you think of this method? Thanks :)

Reply:
I think your approach is good, but I highly recommend following the suggestions provided by django-ninja: https://django-ninja.dev/guides/input/filtering/.